instance/models/mixins/openedx_database.py

```diff
@@ -95,6 +95,10 @@ def mysql_databases(self):
                     }
                 ],
             },
+            {
+                "name": self._get_mysql_database_name("programs"),
+                "user": self._get_mysql_user_name("program"),
+            },
```

Comment:
Ah, good point.  Instances with previously provisioned databases have their [`mysql_provisioned` flag set, and so the databases aren't re-created](https://github.com/open-craft/opencraft/blob/f94660e9e267f85af61ceb50c6600003998b6276/instance/models/mixins/database.py#L144-L148).

It's easy to fix, though.  `provision_mysql()` calls three methods, and only `_create_database` would need to be changed:
* [`_create_database`](https://github.com/open-craft/opencraft/blob/f94660e9e267f85af61ceb50c6600003998b6276/instance/models/mixins/database.py#L69-L74) - could be modified to be [`CREATE DATABASE <database> IF NOT EXISTS`](http://dev.mysql.com/doc/refman/5.7/en/create-database.html). 
* [`create_user`](https://github.com/open-craft/opencraft/blob/f94660e9e267f85af61ceb50c6600003998b6276/instance/models/mixins/database.py#L77-L86) - creates the given user only if it does not already exist.
* [`_grant_privileges`](https://github.com/open-craft/opencraft/blob/f94660e9e267f85af61ceb50c6600003998b6276/instance/models/mixins/database.py#L89-L97) - GRANT is a no-op if the privileges are already granted.

66c50f0 makes this change, but it raises the question: do we need the `mysql_provisioned` flag anymore?